cmd/agent/agent.go

```diff
@@ -86,12 +93,24 @@ func start(cmd *cobra.Command, args []string) {
 		log.Fatal("Failed to create the agent: ", err)
 	}
 
+	consultemplateconfig := &agent.ConsulTemplateConfig{
+		Source:      templatePath,
+		Destination: fmt.Sprintf("%s/%s/%s", configDir, serviceName, METADATAFILE),
+	}
+	runner, err := agent.StartConsulTemplate(a, consultemplateconfig)
+	if err != nil {
+		log.Fatal("could not start consul template: ", err)
+	}
+
```

Comment:
I think this should be encapsulated within `Agent.Start()` instead of being public API.